### DISCUSSION TITLE
Venus 2.91 supported ?

### DISCUSSION THREAD

**Discussion by claib** [Louisvdw/dbus-serialbattery#248] [General]
hi there
very nice job !
Is latest and greatest Venus OS 2.91 supported ?

best
Christoph

**Comment by Louisvdw** [comment]
Yes
The cell voltage ui changes might give problems, but the rest are all working fine.

**Comment by claib** [comment]
THX !
Do you know if it'byting the kwindrem CAN ?

Louis Van Der Walt ***@***.***> schrieb am Mo., 10. Okt.
2022, 19:34:

**Reply by Louisvdw** [reply]
the driver cannot do CAN. (I think that is what you are asking?)

**Comment by Igor-vanlife** [comment]
Hi, found you on Off grid garage youtube channel. I have a problem with installing the driver on venus os 2.92 large. Did try both methods (USB and putty) and fresh venus install on a new sd card. I can't install the driver 

root@raspberrypi4:~# tail -f /data/log/dbus-serialbattery.ttyUSB0/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB0/current': No such file or directory
tail: no files

Any idea what I'm doing wrong??

Thank you so much for your attention and participation.

**Reply by Igor-vanlife** [reply]
Logs as follow:

```bash
root@raspberrypi4:~# wget https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/installrelease.sh
--2022-12-01 12:14:13--  https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/installrelease.sh
Resolving raw.githubusercontent.com... 185.199.109.133, 185.199.108.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 277 [text/plain]
Saving to: ‘installrelease.sh’

installrelease.sh.1 100%[===================>]     277  --.-KB/s    in 0s

2022-12-01 12:14:14 (1.62 MB/s) - ‘installrelease.sh’ saved [277/277]


root@raspberrypi4:~# sh installrelease.sh
cp: omitting directory '/data/etc/dbus-serialbattery/qml'
cp: omitting directory '/data/etc/dbus-serialbattery/service'
root@raspberrypi4:~#
```

And than reboot

**Reply by Louisvdw** [reply]
Hi Igor
Depending on what other devices you have connected the battery driver might not be on ttyUSB0.
You will fist need to look at the serial-starter log file to find the port that is used, and then use that in the serial battery log 
In most cases the problem will be loose connections or adapter problems if the driver does not show any errors in the log file.
https://github.com/Louisvdw/dbus-serialbattery/wiki/Troubleshoot

**Reply by Igor-vanlife** [reply]
Hi, nothing is connected to the RPi yet. 

Just checked this: 

root@raspberrypi4:~# /data/log/serial-starter/current
-sh: /data/log/serial-starter/current: Permission denied

**Reply by Igor-vanlife** [reply]
root@raspberrypi4:~# tail -f /data/log/serial-starter/current
@40000000638898361f08bd84 *** CCGX booted (0) ***
@400000006388983708ebd61c *** starting serial-starter ***
@40000000638898371ba80654 serstart starting
@40000000638898371ca45814 INFO: loading config file /etc/venus/serial-starter.co              nf
@40000000638898380aa71aac INFO: loading config file /data/conf/serial-starter.d
@4000000063889b4a1caa91d4 *** CCGX booted (0) ***
@4000000063889b4b0afe7dc4 *** starting serial-starter ***
@4000000063889b4b19e7eb7c serstart starting
@4000000063889b4b1c59ce0c INFO: loading config file /etc/venus/serial-starter.co              nf
@4000000063889b4c0bc530f4 INFO: loading config file /data/conf/serial-starter.d

**Reply by Louisvdw** [reply]
Use
> tail -f /data/log/serial-starter/current
You will only see something in there is a device is connected that does not already have a driver allocated in the system. So you need to connect at a minimum your USB adapter (even if it is not wired in)

**Reply by Igor-vanlife** [reply]
Ive connected my waveshare usb to rs485 stick to RPi without battery

```bash
root@raspberrypi4:~# tail -f /data/log/serial-starter/current
@4000000063889e8d20d7e7d4 INFO: Start service vedirect-interface.ttyACM0 once
@4000000063889e902e5895fc INFO: Start service dbus-serialbattery.ttyACM0 once
@4000000063889e9c070b7974 INFO: Start service gps-dbus.ttyACM0 once
@4000000063889ea519e7fb1c INFO: Start service vedirect-interface.ttyACM0 once
@4000000063889ea827237284 INFO: Start service dbus-serialbattery.ttyACM0 once
@4000000063889eb33b12b41c INFO: Start service gps-dbus.ttyACM0 once
@4000000063889ebd1223b31c INFO: Start service vedirect-interface.ttyACM0 once
@4000000063889ec01fabd76c INFO: Start service dbus-serialbattery.ttyACM0 once
@4000000063889ecb33e5f514 INFO: Start service gps-dbus.ttyACM0 once
@4000000063889ed50add8074 INFO: Start service vedirect-interface.ttyACM0 once
@4000000063889ed818222b54 INFO: Start service dbus-serialbattery.ttyACM0 once
@4000000063889ee32c6351c4 INFO: Start service gps-dbus.ttyACM0 once
@4000000063889eed0347d224 INFO: Start service vedirect-interface.ttyACM0 once
@4000000063889ef0106fd514 INFO: Start service dbus-serialbattery.ttyACM0 once
@4000000063889efb24580de4 INFO: Start service gps-dbus.ttyACM0 once
^C
root@raspberrypi4:~# tail -f /data/log/dbus-serialbattery.ttyUSB0/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB0/current': No such file or directory
tail: no files
root@raspberrypi4:~# tail -f /data/log/dbus-serialbattery.ttyUSB1/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB1/current': No such file or directory
tail: no files
root@raspberrypi4:~# tail -f /data/log/dbus-serialbattery.ttyUSB2/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB2/current': No such file or directory
tail: no files
root@raspberrypi4:~# tail -f /data/log/dbus-serialbattery.ttyUSB3/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB3/current': No such file or directory
tail: no files
```

**Reply by Louisvdw** [reply]
> INFO: Start service dbus-serialbattery.ttyACM0 once

indicate that as your port. So for the log file you need to use 
> tail -f /data/log/dbus-serialbattery.**ttyACM0**/current

**Reply by Igor-vanlife** [reply]
OK. Now I can see its scanning for bms. 
I have JBD bms with Bluetooth. Ive made a cable to connect (instead of Bluetooth dongle) with 3 cables to my waveshare USB stick. Still no luck to connect it to Venus os. Tried to swap TXD and RXD (A+ B-)  cables and no connection. 
Do I missed something again ?

**Reply by Louisvdw** [reply]
Most likely issue will be with  
> Ive made a cable to connect with 3 cables to my waveshare

The JBD port that the bluetooth is plugged into is a TTL port and RS485 will not work on it. 
You need to use something like https://www.waveshare.com/usb-to-ttl.htm or https://www.waveshare.com/usb-to-ttl-b.htm

**Reply by Igor-vanlife** [reply]
on JBD bms I have just UART port available where Bluetooth is connected. No RS485 port on JBD

**Reply by Igor-vanlife** [reply]
OK. will order a new one and will check. 
Thank you very much for your help and support!

**Reply by Louisvdw** [reply]
Correct. You only have the TTL type UART port available, but the adapter you are trying to use is for RS485. That will not work.
